front/src/applications/osrd/views/OSRDConfig/TimetableSelector.js

```diff
@@ -65,32 +69,48 @@ export default function TimetableSelector(props) {
       </div>
     ));
 
+  useEffect(() => {
+    if (timetableID !== undefined && mustUpdateTimetable) {
+      getTimetable(timetableID);
+      setMustUpdateTimetable(false);
+    } else {
+      setSelectedTimetable(undefined);
+      setMustUpdateTimetable(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mustUpdateTimetable]);
```

Comment:
Please try to find another way than the mustUpdateSomething trigger.  Why not just a useEffect on timetableID ?

Reply:
Can we delay it to a (already planified) refacto of this component ?